src/main/java/codeu/model/store/persistence/PersistentDataStore.java

```diff
@@ -221,13 +232,35 @@ public void writeThrough(Conversation conversation) {
   }
 
   /** Write a Media object to the Datastore service. */
-  public void writeThrough(Media singleMedia) {
+  public void writeThrough(Media singleMedia) throws java.io.IOException {
     Entity mediaEntity = new Entity("chat-media", singleMedia.getId().toString());
     mediaEntity.setProperty("uuid", singleMedia.getId().toString());
     mediaEntity.setProperty("owner_uuid", singleMedia.getOwnerId().toString());
     mediaEntity.setProperty("title", singleMedia.getTitle());
     mediaEntity.setProperty("creation_time", singleMedia.getCreationTime().toString());
-    mediaEntity.setProperty("content", singleMedia.getContent().toString());
+    
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    ImageIO.write(singleMedia.getContent(), "jpg", baos);
+    byte[] contentBytes = baos.toByteArray();
+    String contentString = bytesToHex(contentBytes);
+    Text contentText = new Text(contentString);
+    mediaEntity.setProperty("content", contentText);
+
+    mediaEntity.setProperty("contentType", singleMedia.getContentType());
+    mediaEntity.setProperty("isProfilePicture", singleMedia.getIsProfilePicture().toString());
+    
     datastore.put(mediaEntity);
+
+  }
+
+  private final static char[] hexArray = "0123456789ABCDEF".toCharArray();
+  public static String bytesToHex(byte[] bytes) {
```

Comment:
No existing Java library that does bytes to hex conversion???

Reply:
I believe there are other options that require messing around with pom file and this code is fairly similar to the one that the commons codec library. So far it has been reliable, but I can change it in future iterations.